File: LeetCode/1370.cpp
//Link: https://leetcode.com/problems/increasing-decreasing-string/

class Solution {
public:
    string sortString(string s) {
        int counts[26] = {0}, n = s.length();
        string result;
        for(int i = 0 ; i < n ; i++)
            counts[s[i]-97]++;
        while(n != 0) {
            for(int i = 0 ; i < 26 ; i++){
                if(counts[i] != 0) {
                    result += char(97+i);
                    counts[i]--;
                    n--;
                }
            }
            for(int i = 25 ; i >= 0 ; i--) {
                if(counts[i] != 0) {
                    result += char(97+i);
                    counts[i]--;
                    n--;
                }
            }
        }
        return result;
    }
};

File: LeetCode/py/1370.py
#Link: https://leetcode.com/problems/increasing-decreasing-string/

class Solution:
    def sortString(self, s: str) -> str:
        counts = [0]*26
        result = ''
        n = len(s)
        for ch in s: counts[ord(ch)-97] += 1
        while n != 0:
            for i in range(26):
                if counts[i]:
                    result += chr(97+i)
                    counts[i] -= 1
                    n -= 1
            for i in range(25,-1,-1):
                if counts[i]:
                    result += chr(97+i)
                    counts[i] -= 1
                    n -= 1
        return result
        